File: frontend/src/config/pricing.ts
/**
 * 定价常量（V3.0 商业化方案，与 backend/pricing.py 对齐）
 * 1 字 = 改写 1 个汉字；按实际输出汉字数扣费
 */
export const BENCHMARK_WORDS = 7166;
export const BENCHMARK_YUAN = 0.07;
export const YUAN_PER_WORD = BENCHMARK_YUAN / BENCHMARK_WORDS;

/** 新用户注册赠送 */
export const REGISTRATION_BONUS = 5888;

/** 普通用户每日签到 */
export const DAILY_SIGNIN_GRANT_NORMAL = 888;

/** 看一次完整激励视频（普通用户） */
export const AD_WATCH_REWARD_NORMAL = 2888;

/** 普通用户每日广告上限 */
export const AD_DAILY_LIMIT_NORMAL = 10;

export type MemberTier = "none" | "monthly" | "premium";

export type RechargePackage = {
  yuan: number;
  points: number;
  label: string;
  recommended?: boolean;
  hook?: string;
  /** 字数包永久有效 */
  permanent?: boolean;
};

/** 新用户首充礼包（限购一次） */
export const FIRST_RECHARGE_WORD_PACK = {
  yuan: 2.99,
  points: 18888,
  label: "首充礼包",
  tag: "新用户专享",
  limitOnce: true
} as const;

/** 新用户首充会员包：7 天普通会员体验（限购一次） */
export const FIRST_RECHARGE_MEMBER_PACK = {
  yuan: 2.99,
  days: 7,
  tier: "monthly" as MemberTier,
  label: "首充会员包",
  tag: "体验会员功能",
  limitOnce: true
} as const;

/** 字数包（永久有效） */
export const RECHARGE_PACKAGES: RechargePackage[] = [
  { yuan: 6.99, points: 38888, label: "体验包", hook: "适合课程作业", permanent: true },
  { yuan: 19.9, points: 128888, label: "入门包", hook: "适合课程论文", permanent: true },
  {
    yuan: 39.9,
    points: 388888,
    label: "热门包",
    recommended: true,
    hook: "适合毕业论文",
    permanent: true
  },
  { yuan: 99, points: 1288888, label: "专业包", hook: "适合长期使用", permanent: true }
];

export type MemberPlanDef = {
  tier: MemberTier;
  title: string;
  tag: string;
  tagColor: string;
  yuan: number;
  signinGrant: number;
  adReward: number;
  adDailyLimit: number;
  perks: string[];
};

/** 效率会员体系 */
export const MEMBER_PLANS: MemberPlanDef[] = [
  {
    tier: "monthly",
    title: "普通会员",
    tag: "19.9 元/月",
    tagColor: "blue",
    yuan: 19.9,
    signinGrant: 2888,
    adReward: 3888,
    adDailyLimit: 20,
    perks: [
      "每日签到 2888 字（普通 888）",
      "看广告 3888 字/次（普通 2888）",
      "每日广告 20 次",
      "快速生成模式（多段并行）",
      "会员专属标识"
    ]
  },
  {
    tier: "premium",
    title: "高级会员",
    tag: "39.9 元/月",
    tagColor: "gold",
    yuan: 39.9,
    signinGrant: 3888,
    adReward: 5888,
    adDailyLimit: 30,
    perks: [
      "每日签到 3888 字",
      "看广告 5888 字/次",
      "每日广告 30 次",
      "极速生成模式（更高并行度）",
      "超长文本处理",
      "会员专属标识"
    ]
  }
];

export function signinGrantForTier(tier: MemberTier): number {
  if (tier === "premium") return 3888;
  if (tier === "monthly") return 2888;
  return DAILY_SIGNIN_GRANT_NORMAL;
}

export function adRewardForTier(tier: MemberTier): number {
  if (tier === "premium") return 5888;
  if (tier === "monthly") return 3888;
  return AD_WATCH_REWARD_NORMAL;
}

export function adDailyLimitForTier(tier: MemberTier): number {
  if (tier === "premium") return 30;
  if (tier === "monthly") return 20;
  return AD_DAILY_LIMIT_NORMAL;
}

/** @deprecated V3 字数包为固定档位，保留仅供估算 */
export const POINTS_PER_YUAN = 3000;
export const AD_WATCH_REWARD_POINTS = AD_WATCH_REWARD_NORMAL;
export const DAILY_FREE_GRANT = DAILY_SIGNIN_GRANT_NORMAL;
export const DAILY_FREE_CAP = DAILY_SIGNIN_GRANT_NORMAL;

export function estimateApiYuan(words: number): number {
  return words * YUAN_PER_WORD;
}

export function formatWritableWords(n: number): string {
  return n.toLocaleString("zh-CN");
}

export const HOME_PROMO_COPY = {
  register: `注册送 ${REGISTRATION_BONUS.toLocaleString("zh-CN")} 字`,
  signin: `每日签到 ${DAILY_SIGNIN_GRANT_NORMAL.toLocaleString("zh-CN")} 字`,
  ad: `看广告立即获得 ${AD_WATCH_REWARD_NORMAL.toLocaleString("zh-CN")} 字`,
  firstRecharge: `${FIRST_RECHARGE_WORD_PACK.yuan} 元 · ${FIRST_RECHARGE_WORD_PACK.points.toLocaleString("zh-CN")} 字`,
  hotPack: `${RECHARGE_PACKAGES.find((p) => p.recommended)?.yuan ?? 39.9} 元 · ${(
    RECHARGE_PACKAGES.find((p) => p.recommended)?.points ?? 388888
  ).toLocaleString("zh-CN")} 字`,
  member: "提升签到/广告奖励 · 快速生成"
} as const;

File: backend/pricing.py
"""
定价与商业化常量（V3.0，与 frontend config/pricing.ts 对齐）。

五层变现：免费体验、激励广告、首充礼包、字数包、效率会员。
1 字 = 改写 1 个汉字；按实际输出汉字数扣费。
"""
from __future__ import annotations

import os
from datetime import datetime, timedelta, timezone

BENCHMARK_WORDS = 7166
BENCHMARK_YUAN = 0.07
YUAN_PER_WORD = BENCHMARK_YUAN / BENCHMARK_WORDS

REGISTRATION_BONUS = 5888

MEMBERSHIP_TIER_NONE = "none"
MEMBERSHIP_TIER_MONTHLY = "monthly"
MEMBERSHIP_TIER_PREMIUM = "premium"

SIGNIN_GRANT: dict[str, int] = {
    MEMBERSHIP_TIER_NONE: 888,
    MEMBERSHIP_TIER_MONTHLY: 2888,
    MEMBERSHIP_TIER_PREMIUM: 3888,
}

AD_REWARD: dict[str, int] = {
    MEMBERSHIP_TIER_NONE: 2888,
    MEMBERSHIP_TIER_MONTHLY: 3888,
    MEMBERSHIP_TIER_PREMIUM: 5888,
}

MEMBER_AD_DAILY_LIMIT: dict[str, int] = {
    MEMBERSHIP_TIER_NONE: 10,
    MEMBERSHIP_TIER_MONTHLY: 20,
    MEMBERSHIP_TIER_PREMIUM: 30,
}

MEMBER_RETAIL_YUAN: dict[str, float] = {
    MEMBERSHIP_TIER_MONTHLY: 19.9,
    MEMBERSHIP_TIER_PREMIUM: 39.9,
}

# V3 会员不再按月赠送固定字数池
MEMBER_MONTHLY_POINTS: dict[str, int] = {
    MEMBERSHIP_TIER_MONTHLY: 0,
    MEMBERSHIP_TIER_PREMIUM: 0,
}

FIRST_RECHARGE_WORD_PACK = {"yuan": 2.99, "points": 18_888, "label": "首充礼包"}
FIRST_RECHARGE_MEMBER_DAYS = 7

RECHARGE_PACKAGES: list[dict[str, float | int | str | bool]] = [
    {"yuan": 6.99, "points": 38_888, "label": "体验包", "hook": "适合课程作业"},
    {"yuan": 19.9, "points": 128_888, "label": "入门包", "hook": "适合课程论文"},
    {"yuan": 39.9, "points": 388_888, "label": "热门包", "recommended": True, "hook": "适合毕业论文"},
    {"yuan": 99.0, "points": 1_288_888, "label": "专业包", "hook": "适合长期使用"},
]

# 兼容旧 env / 接口字段名
DAILY_FREE_GRANT = SIGNIN_GRANT[MEMBERSHIP_TIER_NONE]
DAILY_FREE_CAP = SIGNIN_GRANT[MEMBERSHIP_TIER_NONE]


def _normalize_tier(tier: str | None) -> str:
    t = (tier or MEMBERSHIP_TIER_NONE).strip()
    if t in (MEMBERSHIP_TIER_MONTHLY, MEMBERSHIP_TIER_PREMIUM):
        return t
    return MEMBERSHIP_TIER_NONE


def effective_membership_tier(tier: str | None, expires_at: datetime | None) -> str:
    t = _normalize_tier(tier)
    if t == MEMBERSHIP_TIER_NONE:
        return MEMBERSHIP_TIER_NONE
    if expires_at is not None and expires_at <= datetime.now(timezone.utc):
        return MEMBERSHIP_TIER_NONE
    return t


def signin_grant_for_tier(tier: str | None) -> int:
    return SIGNIN_GRANT.get(_normalize_tier(tier), SIGNIN_GRANT[MEMBERSHIP_TIER_NONE])


def ad_reward_for_tier(tier: str | None) -> int:
    env = (os.getenv("AD_WATCH_REWARD_POINTS") or "").strip()
    if env:
        return int(env)
    return AD_REWARD.get(_normalize_tier(tier), AD_REWARD[MEMBERSHIP_TIER_NONE])


def ad_daily_limit_for_tier(tier: str) -> int | None:
    return MEMBER_AD_DAILY_LIMIT.get(_normalize_tier(tier), MEMBER_AD_DAILY_LIMIT[MEMBERSHIP_TIER_NONE])


def member_monthly_grant(tier: str) -> int:
    return MEMBER_MONTHLY_POINTS.get(_normalize_tier(tier), 0)


def ad_watch_reward_points() -> int:
    """默认普通用户广告奖励（创建票据前应按用户 tier 覆盖）。"""
    return ad_reward_for_tier(MEMBERSHIP_TIER_NONE)


def points_per_yuan() -> int:
    return int((os.getenv("POINTS_PER_YUAN") or "3000").strip() or "3000")


def membership_duration_days(tier: str, *, trial: bool = False) -> int | None:
    if trial:
        return FIRST_RECHARGE_MEMBER_DAYS
    if tier in (MEMBERSHIP_TIER_MONTHLY, MEMBERSHIP_TIER_PREMIUM):
        return 30
    return None
